app/src/main/java/com/sun_asterisk/moviedb_49/data/source/remote/MovieRemoteData.kt

```diff
@@ -0,0 +1,17 @@
+package com.sun_asterisk.moviedb_49.data.source.remote
+
+import com.sun_asterisk.moviedb_49.data.model.Movie
+import com.sun_asterisk.moviedb_49.data.source.MovieDBDataSource
+import com.sun_asterisk.moviedb_49.data.source.remote.fetchdata.GetDataJson
+
+class MovieRemoteData : MovieDBDataSource.Remote {
```

Comment:
```suggestion
class MovieRemoteData private constructor() : MovieDBDataSource.Remote {
```

Reply:
```suggestion
object MovieRemoteData : MovieDBDataSource.Remote {
    override fun getData(typeMovie: String, listener: OnFetchDataJsonListener<Movie>) =
        GetDataJson(listener).getDate(typeMovie)
}    
```

Chỉ cần viết như này, lúc sử dụng bạn gọi MovieRemoteData.getData(...) thì nó đã tự có singleton trong đó rồi.